docs/tutorials/authentication.md

```diff
@@ -0,0 +1,71 @@
+---
+title: Authentication and Authorization
+---
+
+Authentication and Authorization
+=============
+
+__Authentication__ is the act of validating that users are whom they claim to be.  
+__Authorization__ is the process of giving the user permission to access a specific resource or function.
+
+The template demo app is set up with OAuth2 authentication and no authorization (Any authenticated user can do anything).
+
+There are 3 main parts that are important to understand in this authentication setup. The OAuth2 authentication server, the SPA web app, and the REST-API.
+
+## The Authentication Server
+
+There are many diffrent _authentication providers_ (Keycloak, Azure, Google, Facebook, etc.), running their own implementation of the OAuth2 protocol. It is these servers responsibility to have the user login and authenticate themselves. The authentication servers authenticate the users against their connected user database (e.g. Azure Active Directory).
+
+If the user authenticate successfully, the auth server returns a __signed JSON Web Token__ (JWT).
+
+In Equinor, Azure AD is the _go-to_ auth provider.
+[This guide](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-register-spa)
+
+## Single Page Web Application
+
+There are no resources to protect in the web app, therefore the job for the SPA in regards to authentication is simple;
+
+  1. Is the user logged in?  
+    a) No - Redirect user to authentication server for authentication  
+    b) Yes - Add the JWT to every request to the API  
+
+Details needed to implement this can be found here; https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow
+
+We use the [react-oauth2-pkce](https://github.com/soofstad/react-oauth2-pkce) library for handling this flow.
```

Comment:
Lot's of reading material for a nooby. 15 minutes of reading that + code is enough though to replicate what is done, even though I don't understand much of what is happening. I guess that is good enough.